docs: add chainlink-integration guide (#388)
Agoric/documentation: main/zoe/guide/contracts/oracle.md

# The Low-level Oracle Query Contract

<Zoe-Version/>

##### [View the code on Github](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/oracle.js) (Last updated: 5-NOV-2020)
##### [View all contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)

**NOTE: You almost certainly do not want to use this contract directly.
Instead, please read the [Chainlink integration
documentation](/chainlink-integration.md)**

This contract lets other contracts or users make single free or fee-based queries to a generic
oracle.  This provides a very
low-level API to issue single queries that an individual off-chain oracle node answers.

> Relying on just a single node can be both expensive and risky. Instead, use the

Reviewer: I think Chainlink already has aggregators on their side.

Author: > I think Chainlink already has aggregators on their side.

They don't on the Agoric chain.  That's implemented in Solidity on Ethereum (and in the higher-level contracts I allude to in Agoric), not in the oracle nodes themselves.

Reviewer: Ah, gotcha. Can we explain this further? I think folks may have the mental model that I have.

Author: > Can we explain this further? I think folks may have the mental model that I have.

Yes, I'll do that.

# The Low-level Oracle Query Contract

<Zoe-Version/>

##### [View the code on Github](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/oracle.js) (Last updated: 5-NOV-2020)
##### [View all contracts on Github](https://github.com/Agoric/agoric-sdk/tree/master/packages/zoe/src/contracts)

**NOTE: You almost certainly do not want to use this contract directly.
Instead, please read the [Chainlink integration
documentation](/chainlink-integration.md)**

> This contract allows other contracts or users to make single queries to a generic
> oracle. The queries can be free, or require a fee.  This provides a very
> low-level API to issue single queries that are answered by an individual
> off-chain oracle node.

Reviewer 2: ```suggestion
This contract lets other contracts or users make single free or fee-based queries to a generic
oracle.  This provides a very
low-level API to issue single queries that an individual off-chain oracle node answers.
```
